pkg/wrangler/context.go

```diff
@@ -0,0 +1,343 @@
+/*
+Package wrangler contains functions for creating a management context with wrangler controllers.
+*/
+package wrangler
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"net/http"
+	"os"
+	"strings"
+	"sync"
+
+	"github.com/rancher/lasso/pkg/controller"
+	"github.com/rancher/lasso/pkg/dynamic"
+	"github.com/rancher/norman/types"
+	managementv3api "github.com/rancher/rancher/pkg/apis/management.cattle.io/v3"
+	"github.com/rancher/rancher/pkg/settings"
+	"github.com/rancher/shepherd/pkg/generated/controllers/management.cattle.io"
+	managementv3 "github.com/rancher/shepherd/pkg/generated/controllers/management.cattle.io/v3"
+	"github.com/rancher/shepherd/pkg/session"
+	"github.com/rancher/shepherd/pkg/wrangler/pkg/generic"
+	"github.com/rancher/wrangler/v2/pkg/apply"
+	"github.com/rancher/wrangler/v2/pkg/generated/controllers/core"
+	corev1 "github.com/rancher/wrangler/v2/pkg/generated/controllers/core/v1"
+	genericwrangler "github.com/rancher/wrangler/v2/pkg/generic"
+	"github.com/rancher/wrangler/v2/pkg/leader"
+	"github.com/rancher/wrangler/v2/pkg/schemes"
+	"github.com/sirupsen/logrus"
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+	"k8s.io/client-go/discovery"
+	"k8s.io/client-go/discovery/cached/memory"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/restmapper"
+	"k8s.io/client-go/tools/clientcmd"
+)
+
+type controllerContextType string
+
+const (
+	Management controllerContextType = "mgmt"
+)
+
+var (
+	localSchemeBuilder = runtime.SchemeBuilder{
+		managementv3api.AddToScheme,
+	}
+	AddToScheme = localSchemeBuilder.AddToScheme
+	Scheme      = runtime.NewScheme()
+)
+
+func init() {
+	metav1.AddToGroupVersion(Scheme, schema.GroupVersion{Version: "v1"})
+	utilruntime.Must(AddToScheme(Scheme))
+	utilruntime.Must(schemes.AddToScheme(Scheme))
+}
+
+type Context struct {
+	RESTConfig *rest.Config
+
+	Apply               apply.Apply
+	Dynamic             *dynamic.Controller
```

Comment:
We don't need dynamic here. We already have our own client so we don't need this.